engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -551,8 +551,8 @@ public void dispose() {
         renderableWorld.dispose();
         worldProvider.dispose();
         // TODO: Shift this to a better place, after a RenderGraph class has been implemented.
-        // TODO: Call ResetDefaultInstance() for every StateChange after it is a method of StateChange.
-        SetViewportToSizeOf.ResetDefaultInstance();
+        // TODO: Call disposeDefaultInstance() for every StateChange after it is a method of StateChange.
```

Comment:
Because, objects should only live for as long as they are useful. The user quits game screen, all objects useful only in game screen should be disposed.